Migrate to an Alpine-based image & introduce a test suite

dduportal opened the pull request:
This Pull Requests follows up #24 .

It introduces:

- A Test suite (using [BATS](https://github.com/sstephenson/bats) ) to validate behaviour of the image
  * I don't cover ALL the use cases. This should be done on the fly ala BDD philosophy
  * We can see this test suite as an auto-generated "tech reference" of the image
- Alpine Linux based image:
  * The final image weight around 470 Mb, which way less than the ~1,55 Gb of the original image
  * asciidoctor Alpine Linux Package is used, but not only
- Asciidoctor-pdf is currently suffering (until next release) from a prawn dependency non wanted behaviour. This PR adds the short-term solution from @mojavelinux . More details here: https://github.com/asciidoctor/asciidoctor-pdf/issues/772
- "Plumbing sugar": gitignore, dockerignore, gitattributes (force unix eol)

- What can be improved (WiP): 
  * [ ] Using the Asciidoctor fixed version for the Alpine package
  * [ ] Using Travis CI (or any other free hosted CI) for building and testing, providing Pull Request feedback and confidence

- Image can be tested from here (build time can be annoying): https://hub.docker.com/r/dduportal/docker-asciidoctor/ (it is a TEMPORARY docker image, do not use it outside testing !)

Now it's your turn to give feedbacks !

dduportal commented:
Travis CI build added. Feel free to stop me here if you do not use Travis for the asciidoctor organization. I can provide Jenkinsfile or Circle CI yml as well.

Build example here: https://travis-ci.org/dduportal/docker-asciidoctor/builds/213795084 

FYI, a good CD pipeline could be:
* Run Travis CI (or any other CI tools) that run the test suite (that takes care of locally building the image)
* If test suite is passing, then, as a post-step, send a webhook to the DockerHub to launch the autobuild.

mojavelinux commented:
> A Test suite

You are my hero already. This is the greatest gift to any project.

mojavelinux commented:
> Asciidoctor-pdf is currently suffering (until next release) from a prawn dependency non wanted behaviour.

I'll have a new release out by the end of the weekend, hopefully even sooner.

mojavelinux commented:
> Feel free to stop me here if you do not use Travis for the asciidoctor organization.

Stop you? I'm going to give you :champagne: :champagne: :champagne:

mojavelinux commented on .gitignore:
```diff
@@ -0,0 +1 @@
+tmp
```
I prefer to make gitignore entries very specific so that they don't catch entries anywhere in the hierarchy. Thus:

```
/tmp/
```

dduportal replied:
👍  I'll update it

mojavelinux commented on Dockerfile:
```diff
@@ -1,61 +1,72 @@
-ENV ASCIIDOCTOR_VERSION "1.5.5"
+ENV JAVA_HOME=/usr/lib/jvm/default-jvm \
+  PATH=${PATH}:${JAVA_HOME}/bin:/fopub/bin \
+  BACKENDS=/asciidoctor-backends \
```
Now that we've made separate gems for reveal.js and bespoke.js, I'm strongly considering dropping the backends from this image. If people disagree, I'm willing to change my mind.

mojavelinux commented on Dockerfile:
```diff
@@ -1,61 +1,72 @@
+  && gem install --no-ri --no-rdoc asciidoctor-revealjs \
   && gem install --no-ri --no-rdoc rake \
   && gem install --no-ri --no-rdoc epubcheck --version 3.0.1 \
   && gem install --no-ri --no-rdoc kindlegen --version 3.0.1 \
```
We should move to kindlegen 3.0.3. I've verified it (on another project) and it works.

dduportal replied:
👍  I'll update it

mojavelinux commented:
I'm going to note some things for the future, but I don't want them to hold up this PR, which is great:

* Consider making a derivative image that adds fopub (aka the DocBook toolchain); we're trying to move away from that and, while there are still people that need it, it should be separate
* I'm not convinced we need to install lazybones (though I do like that we're setting up sdkman)
* Consider adding [Asciidoctor Mathematical](https://github.com/asciidoctor/asciidoctor-mathematical#installation)

bartoszmajsak commented:
That is awesome :) I will grab tmp image and give it a spin for generating @arquillian docs.

bartoszmajsak commented on tests/test_suite.bats:
```diff
@@ -0,0 +1,64 @@
+  docker run -t --rm \
+    -v "${BATS_TEST_DIRNAME}":/documents/ \
+    "${DOCKER_IMAGE_NAME}" \
+      asciidoctor-pdf -D /documents/tmp /documents/fixtures/basic-example.adoc
```
I assume we won't change the content of HTML and PDF. Would it make sense to generate it and use to compare against what docker container produces?

If we see a value in it I can contribute.

dduportal replied:
Hello @bartoszmajsak this totally make sense !

But let's go step by step (I like starting slowly and then improve):
If you're willing to do it now, you can easily fork my fork branch, start your own tests.
THen when you'll have finished, it will be easy to merge on my repo or in this, based on the status of this PR at that moment (power of decentralized tools).

bartoszmajsak replied:
Sure, just wanted to hour your thoughts. It's not for this PR per se, but I will work on it soon.

dduportal replied:
Yup, no problem. Thanks for this, this is a great idea !

jirutka commented on Dockerfile:
```diff
@@ -1,61 +1,72 @@
-    rubygem-nokogiri \
+    jpeg \
+    jpeg-dev \
+    openjdk8 \
```
openjdk8-jre-base should be sufficient here. openjdk8 is complete JDK including dependencies for desktop apps.

dduportal replied:
AFAIK, this come from fopub tool: https://github.com/asciidoctor/asciidoctor-fopub#prerequisites . But given @mojavelinux comment below (https://github.com/asciidoctor/docker-asciidoctor/pull/39#issuecomment-288642823 ) , this would be a great move if we got out the jdk to the jre (even not using java at all in this image)

jirutka commented on Dockerfile:
```diff
@@ -1,61 +1,72 @@
+    jpeg-dev \
+    openjdk8 \
+    patch \
+    python \
```
This package was renamed to python2 (python2-dev).

dduportal replied:
+1 I'll update it

jirutka commented on Dockerfile:
```diff
@@ -1,61 +1,72 @@
+  && mkdir "${BACKENDS}" \
+  && (curl -LkSs https://api.github.com/repos/asciidoctor/asciidoctor-backends/tarball | tar xfz - -C "${BACKENDS}" --strip-components=1) \
+  && ln -s /usr/bin/easy_install-2.7 /usr/local/bin/easy_install \
+  && easy_install 'blockdiag[pdf]' \
```
easy_install is ancient crap that should not be used anymore, use pip instead.

dduportal replied:
👍 I'll update it

dduportal commented:
Based on this first set of feedbacks (Thanks you all !), here is the task list for this PR (I'll take it updated on the fly):

* [ ] `.gitignore` more specific for the tmp dir (and/or use the BATS_TMP_DIR instead)
* [ ] Upgrade kindlegen gem
* [ ] Remove Backends (at least for this PRn, and see if people don't like it)
* [ ] Remove fopub and open an issue for tracking creation of a derivative image (using docker-compose to combine with this one ?) + remove java instead of needed
* [ ]  python:  use the right packages
* [ ] pip instead of easy_install
* [ ] Remove lazybone but keep sdkman installed (not very heavy)
* [ ] Install Asciidoctor mathematical https://github.com/asciidoctor/asciidoctor-mathematical#installation

dduportal commented:
OK, this is all for today :) https://travis-ci.org/dduportal/docker-asciidoctor/builds/214386072 

I got some trouble with asciidoctor-mathematical. Installation is fine, but when running, It was missing liblasem.so . @jirutka I did not find any Alpine package for this sadly. Unless I missed something, this is not possible for now ?

@mojavelinux Is it something required, are shall we do this on another PR ?

bartoszmajsak commented:
Testing with our ARQ docs - works like a charm!

bartoszmajsak commented on Dockerfile:
```diff
@@ -1,61 +1,61 @@
-  && /bin/bash -c "echo sdkman_auto_answer=true > ~/.sdkman/etc/config" \
-  && /bin/bash -c -l "sdk install lazybones"
+  && bash -c "source /root/.sdkman/bin/sdkman-init.sh" \
+  && bash -c "echo sdkman_auto_answer=true > ~/.sdkman/etc/config" \
```
What do we need `sdkman` for now if we don't install `lazybones`?

dduportal replied:
I just removed lazybone following this @mojavelinux comment: https://github.com/asciidoctor/docker-asciidoctor/pull/39#issuecomment-288642823 

But you are right, sdkman does not seems needed now, I also don't see so much value to install it, giving the maintenance overhead. I'm removing it for now, I can still change that later.

dduportal replied:
@bartoszmajsak done, removed SDK. Thanks for this feedback !

bartoszmajsak replied:
![gif-dancing-celebrate-celebration-dance-ecstatic-happy-happy-dance-joy-success-woo-hoo-gif](https://cloud.githubusercontent.com/assets/719616/24289949/8188a89e-1083-11e7-81a9-7d425fddbc06.gif)

jirutka commented:
> It was missing liblasem.so . @jirutka I did not find any Alpine package for this sadly.

I’ve created package for lasem. It’s in the testing repository now. Pnce you verify that it works, I’ll move it to the community to be included in upcoming release v3.6.

Since lasem have many dependencies and it’s not wise to mix stable and edge, I’ve also backported it for v3.5 in my personal repository.

If you want to use it here:

1. Copy https://alpine.geeknet.cz/keys/jakub%40jirutka.cz-56d0d9fd.rsa.pub into this repository, e.g. into `/keys/jakub@jirutka.cz-56d0d9fd.rsa.pub`.
2. Before build copy it into `/etc/apk/keys` inside the image; I guess that `COPY keys/jakub@jirutka.cz-56d0d9fd.rsa.pub /etc/apk/keys/`.
3. Install lasem using: `apk --repository https://alpine.geeknet.cz/packages/v3.5/backports lasem --no-cache add lasem`

I’ll create Alpine package for [mathematical](https://github.com/gjtorikian/mathematical) later.

dduportal commented:
Thanks @jirutka you rocks ! It works now.
This will solves #22 and #13 .

Everything on my side is ok, build is green, image is available in the docker hub:
* https://travis-ci.org/dduportal/docker-asciidoctor/builds/214686367 
* https://hub.docker.com/r/dduportal/docker-asciidoctor/tags/

Final image size is ~461 Mb: 

```bash
docker image ls | grep ascii
docker-asciidoctor                    test                f2b2b8d9c450        41 minutes ago      461 MB
asciidoctor/docker-asciidoctor        latest              30e449946f30        3 weeks ago         1.59 GB
```

@mojavelinux what do you think ?

jirutka commented:
Uff, I’ve finally finished package for ruby-mathematical and its (quite messed) dependencies. It’s currently in edge/testing and backported package for v3.5 is available in my backports repository.

jirutka commented:
You can also install pygments.rb from package backports/ruby-pygments. I guess this is the last problematic dependency.

dduportal commented:
Hello @jirutka @mojavelinux ! Thanks for the packaging work and the feedbacks !

PR updated, tests passing locally, build in progress in Travis: https://travis-ci.org/dduportal/docker-asciidoctor/builds/215154152 

By the way @jirutka  I had to create a symlink in /usr/lib, for `liblasem.so -> liblasem.4.so` to make it work, or I have an error at the execution. WDYT about that (I have no clue on how and where find information on this kind of convention :) )

jirutka commented:
You can get rid of the most explicit dependencies now:

```diff
RUN apk --update --no-cache add \
     asciidoctor="${ASCIIDOCTOR_VERSION}-r0" \
     bash \
-    bison \
     build-base \
-    cairo cairo-dev \
-    cmake \
     curl \
     ca-certificates \
-    flex \
-    gdk-pixbuf gdk-pixbuf-dev \
-    graphviz \
-    jpeg jpeg-dev \
-    libffi libffi-dev \
     libxml2-dev \
-    pango pango-dev \
-    patch \
     python2 python2-dev \
     py-pip \
     ruby \
     ruby-dev \
-    tar \
     ttf-liberation \
     unzip \
     findutils \
     which \
-    wget \
-    zip zlib-dev \
   && apk --repository 'https://alpine.geeknet.cz/packages/v3.5/backports' --no-cache add \
-    lasem \
     ruby-mathematical \
     ruby-pygments \
+    py2-pillow \
```
```diff
   && apk del -r --no-cache \
-    bison \
     build-base \
-    cairo-dev \
-    cmake \
-    flex \
-    gdk-pixbuf-dev \
-    jpeg-dev \
-    libffi-dev \
     libxml2-dev \
-    pango-dev \
     python2-dev \
-    zlib-dev
```

tar and wget are provided by busybox, so they are IMO unnecessary.

Unfortunately it still compiles nokogiri, which is already installed by apk, that’s why libxml2-dev is still needed. The problem is that [gepub](https://rubygems.org/gems/gepub) (dependency of [asciidoctor-epub3](https://rubygems.org/gems/asciidoctor-epub3)) is outdated and requires `nokogiri ~> 1.6.1`, so nokogiri 1.7.1 does not satisfy it. So you will end up with two versions of nokogiri installed. I don’t remember how rubygems handles this situation (when bundler is not used).

I’ll try to test gepub against recent nokogiri.

* * *

py-pip was renamed to pip2-pip:

```diff
-    py-pip \
+    py2-pip \
```

* * *

gem options `--no-ri` and `--no-rdoc` are deprecated and replaced by `--no-document`:

```
$ gem install --help
...
  Deprecated Options:
        --[no-]rdoc                  Generate RDoc for installed gems
                                     Use --document instead
        --[no-]ri                    Generate ri data for installed gems.
                                     Use --document instead
....
```

```diff
-  && gem install --no-ri --no-rdoc prawn --version 2.1.0 \
+  && gem install --no-document prawn --version 2.1.0 \
-  && gem install --no-ri --no-rdoc asciidoctor-epub3 --version 1.5.0.alpha.6 \
+  && gem install --no-document asciidoctor-epub3 --version 1.5.0.alpha.6 \
-  && gem install --no-ri --no-rdoc asciidoctor-pdf --version 1.5.0.alpha.14 \
+  && gem install --no-document asciidoctor-pdf --version 1.5.0.alpha.14 \
-  && gem install --no-ri --no-rdoc epubcheck --version 3.0.1 \
+  && gem install --no-document epubcheck --version 3.0.1 \
-  && gem install --no-ri --no-rdoc kindlegen --version 3.0.3 \
+  && gem install --no-document kindlegen --version 3.0.3 \
-  && gem install --no-ri --no-rdoc asciidoctor-revealjs \
+  && gem install --no-document asciidoctor-revealjs \
-  && gem install --no-ri --no-rdoc asciidoctor-diagram \
+  && gem install --no-document asciidoctor-diagram \
-  && gem install --no-ri --no-rdoc asciidoctor-confluence \
+  && gem install --no-document asciidoctor-confluence \
-  && gem install --no-ri --no-rdoc asciidoctor-mathematical \
+  && gem install --no-document asciidoctor-mathematical \
-  && gem install --no-ri --no-rdoc rake rouge coderay thread_safe slim haml tilt \
+  && gem install --no-document rake rouge coderay thread_safe slim haml tilt \
```

* * *

> By the way @jirutka I had to create a symlink in /usr/lib, for liblasem.so -> liblasem.4.so to make it work, or I have an error at the execution.

This should not be needed anymore, ruby-mathematical takes care of it.

```diff
- && ln -s /usr/lib/liblasem-0.4.so.4 /usr/lib/liblasem.so \
```

I don’t know why the heck lasem upstream puts version number both before and after `.so`, it’s kinda weird and impractical. I’m not sure if it’s less evil to keep it as they want, or rename to liblasem.so.4. However, it’s not a problem when pkg-config is used to discover right flags for linker.

* * *

> apk --update --no-cache add …

The first option is actually named `--update-cache` and as the name suggests, it doesn’t make much sense to use it together with `--no-cache`. Use just `--no-cache`.

* * *

You can group build dependencies under a _virtual package_, so you can easily uninstall them at the end.

```
$ apk add --help
...
  -t, --virtual NAME      Instead of adding all the packages to 'world', create a new virtual package with the listed dependencies and add that to 'world'; the actions of the command are easily reverted by deleting the virtual package
```

```sh
RUN apk add --no-cache \
     asciidoctor="${ASCIIDOCTOR_VERSION}-r0" \
     bash \
     curl \
     ca-certificates \
     python2 \
     ruby \
     ttf-liberation \
     unzip \
     findutils \
     which \
&& apk add --no-cache --virtual .makedepends \
     build-base \
     libxml2-dev \
     python2-dev \
     py2-pip \
     ruby-dev \
```

```diff
-  && apk del -r --no-cache \
-    bison \
-    build-base \
-    cairo-dev \
-    cmake \
-    flex \
-    gdk-pixbuf-dev \
-    jpeg-dev \
-    libffi-dev \
-    libxml2-dev \
-    pango-dev \
-    python2-dev \
-    zlib-dev
+  && apk del -r --no-cache .makedepends
```

jirutka commented on Dockerfile:
```diff
@@ -1,61 +1,70 @@
-ENV BACKENDS /asciidoctor-backends
-ENV GVM_AUTO_ANSWER true
-ENV ASCIIDOCTOR_VERSION "1.5.5"
+ENV JAVA_HOME=/usr/lib/jvm/default-jvm \
```
This can be probably removed since you’ve removed OpenJDK from this image.

dduportal replied:
👍

mgreau commented:
Cool! Good job @dduportal 👍 
It works like a charm with exoplatform developer guide doc. Thanks.

I have another request about the Asciidoctor Docker image, which could be related to this PR. I summarize it here and if you think that it's a subject that should be manage aside, I will create another PR.

So, I would love to be able to execute the following commands:
`docker pull asciidoctor` or at least `docker pull asciidoctor/asciidoctor` and then `docker pull asciidoctor/asciidoctor:1.5.5` `docker pull asciidoctor/backend` ...

My main idea is to:
* keep the `asciidoctor/docker-asciidoctor` image based on fedora and set it as **deprecated**
  * so we don't break all projects that already use it
* **re-organize** this `docker-asciidoctor` GitHub repository to manage several Asciidoctor Docker images (based on Alpine and this [organization example](https://github.com/jessfraz/dockerfiles) 
  * use folders/subfolders to organize Asciidoctor Docker images
  * use scripts and Travis CI to test and deploy Docker images on DockerHub

I use this folders organization to [manage my own Docker images](https://github.com/mgreau/dockerfiles)  and for example to manage my [`mgreau/asciidoctor-bespoke:exo-ioslides`](https://hub.docker.com/r/mgreau/asciidoctor-bespoke/tags/) to develop slides in AsciiDoc.

The benefit would be to easily manage several Asciidoctor Docker images (core, backend, slides...) with tags on DockerHub and in only one GitHub repository.

WDYT?

dduportal commented:
Hello @jirutka , I tried to take in account all your feedbacks, thanks a lot for the effort you put on that, and the time to explain !

Here is the list:

- Removed the JAVA_HOME
- Updated the gem install flags to "--no-document"
- Grouping packages by virtual as you told me (including the cleaning)
- Renaming the PIP package to go "py2-*" and the related test
- Added py2-pillow
- Removed lasem and the symlink: work nice !
- Removed the non needed dependencies
- Corrected the apk flags for cache (I learned again something here, many thanks !)

Image at this state gained more than 100 mb:

```
$ docker images | grep ascii
docker-asciidoctor                    test                   1a9e4e80cde3        About a minute ago   299 MB
```

dduportal commented:
Hello @mgreau , I'll take a look this week on your proposal.

dduportal commented:
Hello @mojavelinux @mgreau @jirutka , do you see anything blocking here, or shall we proceed on mergin' ?

I'll have more time for the next week, if you need any help !


( Ping @mgreau , would you be available to discuss about your proposal in a parallel subject / PR ? )

ckone35 commented:
Hi @dduportal,

thx for this awesome PR,  sadly i can't test it now(https://alpine.geeknet.cz is bocked by my proxy).
I just wondering does graphviz will still be available, I can't see it in the Dockerfile ?

François

jirutka commented:
> https://alpine.geeknet.cz is bocked by my proxy

Why the heck it’s blocked by your proxy? This domain/IP is totally okay, not on any black list I know about.

mgreau commented:
>  ( Ping @mgreau , would you be available to discuss about your proposal in a parallel subject / PR ? )
Yes, sure, I will do it.

> Hello @mojavelinux @mgreau @jirutka , do you see anything blocking here, or shall we proceed on mergin' ?
Nothing blocker on my side, the only thing that I think is important is to tag a version of the current Docker Image version before merging.
Indeed, as we know, it's not a good practice to propose a latest tag. We should improve this and IMO it is the time to do it before updating the distribution, because it's an important upgrade. 
It will be useful to help users if they have problems with this new distribution, they should be able to use this the tag to still use the Fedora base image before we fix problem on the latest image (or another tag for alpine based image).

dduportal commented:
@mgreau You're right, I did not address this important subject.

For this 1st step, breaking the compatibility would not be a good thing. Proposal would be to start with an `alpine`, following the Docker base images pattern (like for openjdk), where the tag embed some application knowledge AND the fact it is based on the alpine distribution as alternative.

Shall we:
* Create a branch `alpine` on the main repo
* Move the current PR target to this branch
* Adapt the DockerHub process to build the branch alpine to the tag `alpine` ?

dduportal commented:
Hello @ckone35 , thanks for this feedback !

I can totally understand the pain of working in a company that blocks randomly URLs from the web with a crappy proxy, and still provide USB ports to the laptops... :)

Don't bother yourself, the image is automatically built on the Dockerhub if you want to try it, instead of having to build it: https://hub.docker.com/r/dduportal/docker-asciidoctor/tags/ .

BtW, do not hesitate to fork, set up an automated build on your behalf on the DockerHub, and your proxy problems would be gone, by delegating the build outside your organisation (and the proxy).

ckone35 commented:
@dduportal, I have reassured you, my usb ports are also blocked
Your idea of delegating is good, i will do it to understand why graphiz is still in your image and not in your dockerfile.

dduportal commented:
@ckone35 Ah ah poor you :)

BtW , my pick is that graphviz is pulled as a dependency as one of the others.

A good PR could be adding a test for this ;)

ovidiub13 commented:
Any plans on merging this to master?

jirutka commented:
@dduportal Alpine v3.6 has been released, so you can remove my repository, all needed packages are in Alpine aports now:

```diff

- FROM alpine:3.5
+ FROM alpine:3.6

- ADD https://alpine.geeknet.cz/keys/jakub%40jirutka.cz-56d0d9fd.rsa.pub /etc/apk/keys/

     findutils \
     which \
+    py2-pillow \
+    ruby-mathematical \
+    ruby-pygments \
   && apk add --no-cache --virtual .makedepends \
     build-base \
-  && apk --repository 'https://alpine.geeknet.cz/packages/v3.5/backports' --no-cache add \
-    py2-pillow \
-    ruby-mathematical \
-    ruby-pygments \
```

dduportal commented:
Hello, thanks for the notice @jirutka  !

It has been updated to alpine 3.6, with Gems updates also (asciidoctor pdf and asciidocotr alpine packer to r1).

Ping @mojavelinux if it is ok for you to merge ?

jirutka commented:
@mojavelinux Why is this still not merged?

ovidiub13 commented:
@jirutka I've been wondering the same thing for months

wimdeblauwe commented:
I would love to see this happen as well. Seems the current version on dockerhub is still using asciidoctor 1.5.5 (https://hub.docker.com/r/dduportal/docker-asciidoctor/~/dockerfile/).

wimdeblauwe commented:
Just a heads up that I tried the image but I get the following error if I want to use asciidoctor diagram:

     asciidoctor-diagram: ERROR: Failed to generate image: Could not find Java executable

I used the latest version of the code in this PR + I replaced version `3.6` with `edge` for the Linux and set asciidoctor version to `1.5.6.1-r0`

jirutka commented:
> I get the following error if I want to use asciidoctor diagram:
>
> ```
> asciidoctor-diagram: ERROR: Failed to generate image: Could not find Java executable
> ```

Well, that’s no surprising, there’s no `openjdk-jre-base` package in the dockerfile. However, why does it need Java? o.O

> I replaced version `3.6` with `edge`…

That’s not a good idea, `edge` is unstable.

dduportal commented:
Hello @wimdeblauwe @jirutka! Thanks for your feedbacks!
About the missing java in the image, it comes from  @mojavelinux 
 comment at the beginning of the current PR: https://github.com/asciidoctor/docker-asciidoctor/pull/39#discussion_r107604730 
=> One strategy would be having a dedicated tag or image with java embedded to cover this. The tradeoff here is that a lot of the user might not need the java, but are interested in a lightweight image. But still, in term of usability, the end-user should not have to care about this.
=> Maybe a composed image could be a solution, with the new https://docs.docker.com/engine/userguide/eng-image/multistage-build/ .

Ping @mojavelinux we need you or a maintainer to merge this in order to iterate. I am willing to help, even maintaining if you need help or time.

jirutka commented:
@dduportal I pinged @mojavelinux [on Twitter](https://twitter.com/jakubjirutka/status/916278446837370885) and asked him to give you a write access. It really sucks to wait months for response/approval.

dduportal commented:
Thanks @jirutka . I would not say that "it sucks" because I already had issues to cover all requests on my very tiny personnal projects, so I can understand the overloading for @mojavelinux .

By the way, willing to maintain will help I hope :)

mojavelinux commented:
It's definitely time (maybe even past time) to make this switch. We're all ready for it and other requests are getting backed up because this isn't merged yet. While I could merge, I would only just be pressing a button, not testing it. Therefore, I'm granting @dduportal admin rights on the repository. @dduportal you can choose to maintain it or grant permission to another trust member of the community willing to step forward. You've demonstrated your skill set in this issue and I trust you to be the one who merges it. We all appreciate your contribution. I want to empower you to continue this work.

dduportal commented:
Thanks @mojavelinux for this trust.

Just tested again and it is working on my Mac, on Travis, and on my remote Ubuntu server.

Now merging and taking care of any issue question from now.

mojavelinux commented:
:clap:
